src/main/kotlin/no/nav/familie/ef/sak/api/beregning/BeregningService.kt

```diff
@@ -1,17 +1,133 @@
+        return beløpForInnteksperioder.mapNotNull {
+            if (it.fraOgMedDato.isEqualOrAfter(vedtaksperiodeFraOgmedDato) && it.tilDato.isEqualOrBefore(vedtaksperiodeTilDato)) {
+                it
+            } else if (it.starterFørVedtaksperiodeOgOverlapper(vedtaksperiodeFraOgmedDato, vedtaksperiodeTilDato)) {
+                it.copy(fraOgMedDato = vedtaksperiodeFraOgmedDato)
+            } else if (it.starterEtterVedtaksperiodeOgOverlapper(
+                            vedtaksperiodeFraOgmedDato,
+                            vedtaksperiodeTilDato)) {
+                it.copy(tilDato = vedtaksperiodeTilDato)
+            } else if (it.starterFørOgSlutterEtterVedtaksperiode(vedtaksperiodeFraOgmedDato,
+                                                                 vedtaksperiodeTilDato)) {
+                it.copy(tilDato = vedtaksperiodeTilDato, fraOgMedDato = vedtaksperiodeFraOgmedDato)
+            } else {
+                null
+            }
+        }
+    }
+
+
+    private fun validerVedtaksperioder(vedtaksperiode: List<Periode>) {
+        val sorterteVedtaksperioder = vedtaksperiode.sortedBy { it.fradato }
+
+        sorterteVedtaksperioder.forEachIndexed { index, periode ->
+            if (sorterteVedtaksperioder.size > 1 && index < sorterteVedtaksperioder.lastIndex) {
+                feilHvis(periode.tildato.isEqualOrAfter(sorterteVedtaksperioder[index + 1].fradato)) {
+                    "Vedtaksperioder ${vedtaksperiode} overlapper"
+                }
+            }
+        }
+    }
+
+    private fun validerInnteksperioder(inntektsperioder: List<Inntektsperiode>, vedtaksperiode: List<Periode>) {
+        feilHvis(inntektsperioder.isEmpty()) {
+            "Inntektsperioder kan ikke være tom liste"
+        }
+
+        val sorterteIntekter = inntektsperioder.sortedBy { it.startDato }
+        val sorterteVedtaksperioder = vedtaksperiode.sortedBy { it.fradato }
+
+
+        sorterteIntekter.forEachIndexed { index, inntektsperiode ->
+            if (index === 0) {
+                feilHvis(!inntektsperiode.startDato.isEqualOrBefore(sorterteVedtaksperioder.first().fradato)) {
+                    "Inntektsperioder ${inntektsperioder} begynner etter vedtaksperioder ${vedtaksperiode}"
+                }
+            }
+            if (index === sorterteIntekter.lastIndex) {
+                feilHvis(!inntektsperiode.sluttDato.isEqualOrAfter(sorterteVedtaksperioder.last().tildato)) {
+                    "Inntektsperioder ${inntektsperioder} slutter før vedtaksperioder ${vedtaksperiode} "
+                }
+            }
+            if (sorterteIntekter.size > 1 && index < sorterteIntekter.lastIndex) {
+                feilHvis(inntektsperiode.startDato.isEqual(sorterteIntekter[index + 1].startDato)) {
+                    "Inntektsperioder ${inntektsperioder} overlapper eller er ikke sammenhengde for vedtaksperioder ${vedtaksperiode}"
+                }
+            }
```

Comment:
Denne går å gjøre litt "enklere" hvis man ønsker

```suggestion
        feilHvis(!sorterteIntekter.first().startDato.isEqualOrBefore(sorterteVedtaksperioder.first().fradato)) {
            "Inntektsperioder $inntektsperioder begynner etter vedtaksperioder $vedtaksperiode"
        }
        feilHvis(!sorterteIntekter.last().sluttDato.isEqualOrAfter(sorterteVedtaksperioder.last().tildato)) {
            "Inntektsperioder $inntektsperioder slutter før vedtaksperioder $vedtaksperiode "
        }
        feilHvis(inntektsperioder.zipWithNext { a, b -> a.startDato < b.startDato }.any { !it }) {
            "Inntektsperioder $inntektsperioder overlapper eller er ikke sammenhengde for vedtaksperioder $vedtaksperiode"
        }
```

Reply:
Enig, fikset  [30a9237](https://github.com/navikt/familie-ef-sak/pull/630/commits/30a92379601c378b835a2d8cbd2ce634f66a48c4)